File: Day252/sortArrayByParity.go
package main 

import (
	"fmt"
)

func sortArrayByParity(nums []int) []int {
	point := 0
	for i:=0; i<len(nums); i++ {
		if nums[i]&1 != 1 {
			nums[i], nums[point] = nums[point], nums[i]
			point++
		}
	}
	return nums
}

func main() {
	nums := []int{3,1,2,4}
	fmt.Println(sortArrayByParity(nums))
}

File: Day252/sortArrayByParity.java
package Day252;

import java.util.Arrays;

public class sortArrayByParity{

    public static int[] sabp (int nums[]) {
        int point = 0;
        for (int i=0; i<nums.length; i++) {
            if ((nums[i] & 1) != 1) {
                int temp = nums[point];
                nums[point++] = nums[i];
                nums[i] = temp;
            }
        }
        return nums;
    }

    public static void main(String[] args) {
        int nums [] = {3,1,2,4};
        System.out.println(Arrays.toString(sabp(nums)));
    } 
}